internal/pkg/manifest/workload.go

```diff
@@ -430,17 +447,87 @@ func (p *PlacementArgOrString) UnmarshalYAML(value *yaml.Node) error {
 	return nil
 }
 
-// PlacementArgs represents what subnets to place tasks.
+// PlacementArgs represents where to place tasks.
 type PlacementArgs struct {
-	Subnets []string `yaml:"subnets"`
+	Subnets SubnetListOrArgs `yaml:"subnets"`
 }
 
 func (p *PlacementArgs) isEmpty() bool {
-	return p.Subnets == nil || len(p.Subnets) == 0
+	return p.Subnets.isEmpty()
 }
 
-// PlacementString represents what types of subnets (public or private subnets) to place tasks.
-type PlacementString string
+// SubnetListOrArgs represents what subnets to place tasks. It supports unmarshalling
+// yaml which can either be of type SubnetArgs or a list of strings.
+type SubnetListOrArgs struct {
+	IDs []string
+	SubnetArgs
+}
+
+func (s *SubnetListOrArgs) isEmpty() bool {
+	return len(s.IDs) == 0 && s.SubnetArgs.isEmpty()
+}
+
+type dynamicSubnets struct {
```

Comment:
i'm not seeing where this gets used...

Reply:
https://github.com/aws/copilot-cli/pull/3727/files#diff-6272e76645f1b500c7be4973be9e3d455c5f1aeb5350aea1890782dfec9eba20R60